pkg/kubelet/pleg/evented.go

```diff
@@ -93,7 +93,7 @@ func NewEventedPLEG(runtime kubecontainer.Runtime, runtimeService internalapi.Ru
 		runtimeService:              runtimeService,
 		eventChannel:                eventChannel,
 		cache:                       cache,
-		genericPleg:                 genericPleg,
+		genericPleg:                 genericPleg.(podLifecycleEventGeneratorHandler),
```

Comment:
Just for the sake of safety I'd suggest to add a casting check, e.g:
```
--- a/pkg/kubelet/pleg/evented.go
+++ b/pkg/kubelet/pleg/evented.go
@@ -87,17 +87,21 @@ type EventedPLEG struct {
 // NewEventedPLEG instantiates a new EventedPLEG object and return it.
 func NewEventedPLEG(runtime kubecontainer.Runtime, runtimeService internalapi.RuntimeService, eventChannel chan *PodLifecycleEvent,
        cache kubecontainer.Cache, genericPleg PodLifecycleEventGenerator, eventedPlegMaxStreamRetries int,
-       relistDuration *RelistDuration, clock clock.Clock) PodLifecycleEventGenerator {
+       relistDuration *RelistDuration, clock clock.Clock) (PodLifecycleEventGenerator, error) {
+       handler, ok := genericPleg.(podLifecycleEventGeneratorHandler)
+       if !ok {
+               return nil, fmt.Errorf("%v doesn't implement podLifecycleEventGeneratorHandler interface", genericPleg)
+       }
        return &EventedPLEG{
                runtime:                     runtime,
                runtimeService:              runtimeService,
                eventChannel:                eventChannel,
                cache:                       cache,
-               genericPleg:                 genericPleg,
+               genericPleg:                 handler,
                eventedPlegMaxStreamRetries: eventedPlegMaxStreamRetries,
                relistDuration:              relistDuration,
                clock:                       clock,
-       }
+       }, nil
 }

--- a/pkg/kubelet/kubelet.go
+++ b/pkg/kubelet/kubelet.go
@@ -734,8 +734,11 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
                        RelistPeriod:    genericPlegRelistPeriod,
                        RelistThreshold: genericPlegRelistThreshold,
                }
-               klet.eventedPleg = pleg.NewEventedPLEG(klet.containerRuntime, klet.runtimeService, eventChannel,
+               klet.eventedPleg, err = pleg.NewEventedPLEG(klet.containerRuntime, klet.runtimeService, eventChannel,
                        klet.podCache, klet.pleg, eventedPlegMaxStreamRetries, eventedRelistDuration, clock.RealClock{})
+               if err != nil {
+                       return nil, err
+               }
        } else {
                genericRelistDuration := &pleg.RelistDuration{
                        RelistPeriod:    genericPlegRelistPeriod,
```

Reply:
@bart0sh updated. Thanks.